lib/peep.rb

```diff
@@ -8,7 +9,22 @@ def initialize(text)
   end
 
   def self.all
-    [Peep.new('peep1'), Peep.new('peep2')]
+    @connection = connection
+    rs = @connection.exec( "SELECT text FROM peeps" )
+
+    puts rs
+
+    rs.map { |peep| Peep.new(peep['text']) }
+  end
+
+  private
```

Comment:
i like how you have used the private function to control access to your methods. The use of => to access the DB is very interesting too